tests/tests_files_elasticsearch_certs_incomplete.yml

```diff
@@ -0,0 +1,52 @@
+- name: Error case for Elasticsearch config - cert and ca_cert_src are missing
+  hosts: all
+  become: true
+  vars:
+    __test_inputfiles_dir: /var/log/inputdirectory
+    __test_inputfiles_conf: /etc/rsyslog.d/90-input-files-files_input.conf
+    __test_outputfiles_conf: /etc/rsyslog.d/31-output-elasticsearch-elasticsearch_output.conf
+    __test_ca_cert: /tmp/es-ca.crt
+    __test_cert: /tmp/es-cert.pem
+    __test_key: /tmp/es-key.pem
+
+  tasks:
+    - name: generate test files
+      copy:
+        dest: "{{ item }}"
+        content:
+          This is a fake {{ item }}.
+      delegate_to: localhost
+      loop:
+        - "{{ __test_ca_cert }}"
+        - "{{ __test_cert }}"
+        - "{{ __test_key }}"
+
+    - block:
+      - name: deploy config to send to elasticsearch
+        vars:
+          logging_outputs:
+            - name: elasticsearch_output
+              type: elasticsearch
+              server_host: logging-es
+              server_port: 9200
+              index_prefix: project.
+              input_type: ovirt
+              retryfailures: off
+              ca_cert: /etc/rsyslog.d/ca_cert.crt
+              key: /etc/rsyslog.d/key.pem
+              cert_src: "{{ __test_cert }}"
+              key_src: "{{ __test_key }}"
+          logging_inputs:
+            - name: files_input
+              type: files
+              rsyslog_input_log_path: "{{ __test_inputfiles_dir }}/*.log"
+          logging_flows:
+            - name: flow_0
+              inputs: [files_input]
+              outputs: [elasticsearch_output, elasticsearch_output_ops]
+        include_role:
+          name: linux-system-roles.logging
+        register: result
+      rescue:
+        - debug:
+            msg: "Caught an expected error - {{ result }}"
```

Comment:
Does this mask the error?  That is, will the test report an error here, or do you need a separate `assert` or `fail` task?
https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html#blocks-error-handling
I'm not sure if `result` set in the `include_role` task in the `block` will be available in the `rescue` section.  You can use `ansible_failed_result` to get the result of the failed task.  Note that the page refers to `ansible_failed_task` but there is currently a bug and you cannot use this variable.

Reply:
Thank you, @richm. That's exactly what I was looking for.
Also, rebased.